src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static increase(animal) {
+    this.alive.push(animal);
+  }
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.increase(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.increase(this);
+  }
+
+  bite(beast) {
+    if (beast.hidden === true || beast instanceof Carnivore) {
+      return beast.health;
+    } else {
+      beast.health -= 50;
+    }
+
+    if (beast.health === 0) {
+      const dead = Animal.alive.indexOf(beast);
```

Comment:
What if `indexOf` returns -1, what will `splice` delete ?

We need specific check